src/core/pointcloud/qgspointcloudlayerexporter.h

```diff
@@ -0,0 +1,339 @@
+     */
+    void setLayerName( const QString &name ) { mName = name; };
+
+    /**
+     * Gets the name for the new layer.
+     */
+    QString layerName() const { return mFilename; };
+
+    /**
+     * Sets a filter extent for points to be exported in the target CRS
+     * Points that fall outside the extent will be skipped.
+     * \see setCrs()
+     */
+    void setFilterExtent( const QgsRectangle extent ) { mExtent = extent; };
+
+    /**
+     * Gets the filter extent for points to be exported.
+     */
+    QgsRectangle filterExtent() const { return mExtent; };
+
+    /**
+     * Sets an inclusive range for Z values to be exported.
+     * Points with Z values outside the range will be skipped.
+     */
+    void setZRange( const QgsDoubleRange zRange ) { mZRange = zRange; };
+
+    /**
+     * Gets the inclusive range for Z values to be exported.
+     */
+    QgsDoubleRange zRange() const { return mZRange; };
+
+    /**
+     * Sets a QgsFeedback object to allow cancellation / progress reporting.
+     *
+     * \note The \a feedback object must exist for the lifetime of the exporter.
+     */
+    void setFeedback( QgsFeedback *feedback ) { mFeedback = feedback; };
+
+    /**
+     * Gets a pointer to the QgsFeedback object used for cancellation / progress reporting, or nullptr if not set.
+     */
+    QgsFeedback *feedback() const { return mFeedback; };
+
+    /**
+     * Sets the list of point cloud \a attributes that will be exported.
+     * If never called, all attributes will be exported.
+     * \note This has no effect when exporting to LAZ format.
+     */
+    void setAttributes( const QStringList &attributes );
+
+    /**
+     * Sets that no attributes will be exported.
+     * \note This has no effect when exporting to LAZ format.
+     */
+    void setNoAttributes() { mRequestedAttributes.clear(); };
+
+    /**
+     * Sets that all attributes will be exported.
+     */
+    void setAllAttributes();
+
+    /**
+     * Gets the list of point cloud attributes that will be exported.
+     */
+    QStringList attributes() const { return mRequestedAttributes; };
+
+    /**
+     * Sets the \a crs for the exported file, and the transform \a context that will be used for
+     * for reprojection if different from the point cloud layer's CRS.
+     */
+    void setCrs( const QgsCoordinateReferenceSystem &crs, const QgsCoordinateTransformContext &context = QgsCoordinateTransformContext() ) { mTargetCrs = crs; mTransformContext = context; };
+
+    /**
+     * Gets the \a crs for the exported file.
+     */
+    QgsCoordinateReferenceSystem crs() const { return mTargetCrs; };
+
+    /**
+     * Sets the \a format for the exported file.
+     * \returns true if the \a format is supported, false otherwise.
+     * \see supportedFormats()
+     */
+    bool setFormat( const QString &format );
+
+    /**
+     * Returns the format for the exported file or layer.
+     */
+    QString format() const { return mFormat; };
```

Comment:
What do you think of having an enum defined with possible formats rather than using a QString?

Reply:
Hmm, that's not a bad thought, it will make clearer what the supported formats are.
Should we skip the `supportedFormats()` method then?

I will address this in an upcoming PR

Reply:
yeah probably we can skip supportedFormats() from the API